### PR TITLE
PR. User management

### LINKED ISSUE #22: User management
Add new REST API methods to get, create, update, delete users.
The API should be available for “admin” only. 
All projects of removed user should be removed too.

### PR DESCRIPTION
**Pull request by tempo92**
Closes #22

### CODE REVIEW

File: src/main/java/org/tempo/springEdu/service/UserService.java [hunk 1]
```diff
@@ -1,29 +1,76 @@
...
+import java.util.stream.Collectors;
 
 @Component
 public class UserService {
 
     @Autowired
     private UserRepository userRepository;
 
+    @Autowired
+    private ProjectRepository projectRepository;
```
Comment (iv21dk) [inline src/main/java/org/tempo/springEdu/service/UserService.java:31]: I suggest to use `ProjectService` instead

**Review by iv21dk — CHANGES_REQUESTED** [review]

File: src/main/java/org/tempo/springEdu/service/UserService.java [hunk 2]
```diff
@@ -1,29 +1,76 @@
...
+        createDefaultUserIfNeed("user", "user", RoleName.USER);
+        createDefaultUserIfNeed("admin", "admin", RoleName.ADMIN);
+    }
+
+    public List<UserDto> findAllDto() {
+        return entityListToDtoList(
+                userRepository.findAll());
+    }
+
+    public void create(UserUpdateDto userUpdateDto, User user) {
```
[the remark this answers is not in the document]
Comment (tempo92) [reply src/main/java/org/tempo/springEdu/service/UserService.java]: Fixed

File: src/main/java/org/tempo/springEdu/service/UserService.java (hunk 1 above)
Comment (tempo92) [reply src/main/java/org/tempo/springEdu/service/UserService.java]: Fixed

File: src/main/java/org/tempo/springEdu/service/UserService.java [hunk 3]
```diff
@@ -1,29 +1,76 @@
...
+        userRepository.delete(delUser);
+    }
+
+    public User findById(String id) {
+        return userRepository.findById(id)
+                .orElseThrow(() -> new ObjectNotFoundException(
+                        String.format("User with id=%s not found", id)));
+    }
+
+    public void update(String id, UserUpdateDto dto, User user) {
```
[the remark this answers is not in the document]
Comment (tempo92) [reply src/main/java/org/tempo/springEdu/service/UserService.java]: Fixed

**Pull request merged**